rosidl_generator_dds_idl/cmake/rosidl_generate_dds_interfaces.cmake

```diff
@@ -40,21 +40,33 @@ macro(rosidl_generate_dds_interfaces target)
       "unused arguments: ${_ARG_UNPARSED_ARGUMENTS}")
   endif()
 
-  set(_output_path "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_dds_idl/${PROJECT_NAME}")
-  foreach(_subfolder ${_ARG_OUTPUT_SUBFOLDERS})
-    set(_output_path "${_output_path}/${_subfolder}")
-  endforeach()
-  set(_generated_files "")
+  set(_output_basepath "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_dds_idl/${PROJECT_NAME}")
+  set(_generated_msg_files "")
+  set(_generated_srv_files "")
   foreach(_idl_file ${_ARG_IDL_FILES})
     get_filename_component(_extension "${_idl_file}" EXT)
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    set(_output_path "${_output_basepath}/${_parent_folder}")
+    foreach(_subfolder ${_ARG_OUTPUT_SUBFOLDERS})
+      set(_output_path "${_output_path}/${_subfolder}")
+    endforeach()
     get_filename_component(_name "${_idl_file}" NAME_WE)
     if("${_extension}" STREQUAL ".msg")
-      list(APPEND _generated_files "${_output_path}/${_name}_.idl")
+      if("${_parent_folder} " STREQUAL "msg ")
+        list(APPEND _generated_msg_files "${_output_path}/${_name}_.idl")
+      elseif("${_parent_folder} " STREQUAL "srv ")
+        list(APPEND _generated_srv_files "${_output_path}/${_name}_.idl")
+      else()
+        message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
+      endif()
     elseif("${_extension}" STREQUAL ".srv")
-      list(APPEND _generated_files "${_output_path}/Sample${_name}Request_.idl")
-      list(APPEND _generated_files "${_output_path}/Sample${_name}Response_.idl")
-      list(APPEND _generated_files "${_output_path}/WriteSample${_name}Request_.idl")
-      list(APPEND _generated_files "${_output_path}/WriteSample${_name}Response_.idl")
+      # TODO this is only done for opensplice
```

Comment:
What does the `TODO(name)` imply? That <name> is going to address the to do?

What name would you suggest in this case then?

* mine because I added the comment to move the code?
* @esteve since he added the `Sample` code there in the first place?